nixos/doc/manual/configuration/wireless.xml

```diff
@@ -32,6 +32,33 @@
   wpa_supplicant.conf(5)).
  </para>
 
+ <para>
+  When using a network with
+  <link xlink:href="https://wiki.archlinux.org/index.php/Wireless_network_configuration#WPA2_Enterprise">WPA2 Enterprise</link>
+  for authentication, it's possible to configure it declaratively using the <literal>auth</literal> option:
+<programlisting>
+<xref linkend="opt-networking.wireless.networks" /> = {
+  wpa2-enterprise-net = {
+    auth = ''
+      key_mgmt=WPA-WAP
+      eap=PEAP
+      identity="user@example.com"
+      password="secret"
+    '';
+  };
+}
+</programlisting>
+  <emphasis>Note: please keep in mind that the configuration file will be written to a store path
+  which is world-readable!</emphasis>
+
+  To avoid leaking your cleartext-password, <literal>wpa_supplicant.conf</literal> allows to
+  specify a hashed password using a configuration like this: <literal>password=hash:{your-hashed-password}</literal>.
+  Such a password can be a an NTLM password which can be generated with the following command:
+<programlisting>
+echo -n $YOURPASSWORD | iconv -t utf16le | openssl md4
```

Comment:
How security does one gain from `md4`?

Reply:
As far as I've seen it for now, this is currently the only possibility to provide hashed passwords. I just had a look at the code and unfortunately it seems as this is the only way to hash passwords and I guess that we should mention this (I just found most of the information from stackexchange threads, the Arch wiki and some playing around though).

Currently I can imagine two problematic cases:

* A compromised Nix store: as long as Nix doesn't implement encryption capabilities for store paths the leaked configuration file is sufficient to access a WPA2 enterprise network using the victim's credentials. I do think that this can be helpful though as in several cases the users authenticated using LDAP, so the cleartext password won't be leaked (as mentioned in the previous sentence in the change, but the more I think about it the more I think that this should be mentioned in an `emphasis` in a separate paragraph.)

* The `md4` hash itself. Unfortunately I didn't find a suitable alternative and I don't know the codebase well enough to efficiently (and safely!) fix this upstream. I'd say that the measure I'm mentioning here makes it harder to leak the cleartext password, but doesn't add tru security which has to be kept in mind when using this feature.

My original intention was to document that our `wireless` module allows connections to WPA-EAP networks as well (I wanted to try this out for a given network yesterday, looked into the manual where this was documented and didn't find it).

I guess that we should mention the hashing approach (there aren't too much sources mentioning it ATM), but I guess that I should make it **way** more explicit in my wording that this still contains risks.

Reply:
Yes, we should not advertise md4 as a secure alternative that can protect your password in the nix store, when this is not the case. The example itself is good though.